### ISSUE TITLE
Allow omission of `project.dependencies`

### ISSUE THREAD

**Issue by ofek** [repo-helper/hatch-requirements-txt#21]
## Description

https://github.com/aws/sagemaker-python-sdk can't migrate over yet because only their `project.optional-dependencies` are dynamic

cc @maresb

**Comment by maresb** [comment]
Ah, ok, I was confused at first... if I understand correctly, they want to use dynamic `optional-dependencies` without dynamic `dependencies`:

* [Normal dependencies](https://github.com/aws/sagemaker-python-sdk/blob/master/setup.py#L48-L62)
* [Dynamic optional dependencies](https://github.com/aws/sagemaker-python-sdk/blob/master/setup.py#L64-L73)

**Comment by maresb** [comment]
Very quickly taking a look, it seems like the problematic logic is [here](https://github.com/repo-helper/hatch-requirements-txt/blob/bd199f5cfadcecb7f67894757e8d04e92b50427b/hatch_requirements_txt/__init__.py#L126-L131). (I haven't thought this through, I might be confused.) In the first `if` we assume that if the plugin is loaded then ordinary dependencies are not allowed. But really we want a logic block like [this](https://github.com/repo-helper/hatch-requirements-txt/blob/bd199f5cfadcecb7f67894757e8d04e92b50427b/hatch_requirements_txt/__init__.py#L133-L140) around [the part that handles dependencies](https://github.com/repo-helper/hatch-requirements-txt/blob/bd199f5cfadcecb7f67894757e8d04e92b50427b/hatch_requirements_txt/__init__.py#L110-L124).

In other words, loading the plugin should enable but not enforce `dependencies` to be specified dynamically.
